pgdog/src/frontend/client/mod.rs

```diff
@@ -444,24 +447,21 @@ impl Client {
                 Some(Command::StartTransaction(query)) => {
                     if let BufferedQuery::Query(_) = query {
                         self.start_transaction().await?;
-                        inner.start_transaction = Some(query.clone());
```

Comment:
This is important to record. `BEGIN` can have multiple arguments that affect what kind of transaction we're starting: https://www.postgresql.org/docs/current/sql-begin.html